public/app/features/dimensions/types.ts

```diff
@@ -66,6 +66,7 @@ export interface ColorDimensionConfig extends BaseDimensionConfig<string> {}
 /** Places that use the value */
 export interface ResourceDimensionOptions {
   resourceType: 'icon' | 'image';
+  folderIndex: number;
```

Comment:
rather than index... maybe have the folder name?  

Also `showSourceRadio` should be here not in the saved model